main.go

```diff
@@ -0,0 +1,35 @@
+package storageconsul
+
+import (
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/certmagic"
+)
+
+type TLSConsul struct {
+	storage *ConsulStorage
+}
+
+func init() {
+	caddy.RegisterModule(TLSConsul{})
```

Comment:
```suggestion
	caddy.RegisterModule(&TLSConsul{})
```
If you use a pointer here you can also use pointer receiver for all methods in an uniform way (and that's what I would prefer in that case).

Reply:
Pretty sure this doesn't need to be a pointer. But, it's up to you I guess.

Reply:
No, not necessarily - it's only me and my inner Monk 😉
I prefer not to mix and match pointer and value receiver functions on the same struct. For me this feels better and more elegant. 
One function, `Provision()` changes `TLSConsul` by adding the `consulStorage` instance so it *has* to be a pointer receiver. And because of that and my strange logic all three functions should get pointer receivers. 
And if `CaddyModule() caddy.ModuleInfo` has a pointer receiver too I need to provide the pointer to in this place to make this work with `caddy.RegisterModul`.

Reply:
Not gonna lie, I looked at the core modules inside Caddy's code to help me implement the new module interface 😃
A mix of both pointers & values is used on CaddyModule() & Provision() there. Here's the [filewriter ](https://github.com/caddyserver/caddy/blob/master/modules/logging/filewriter.go) module for reference.
Either we normalize and follow Caddy's core module' style or we use pointers/receivers everywhere. I'm fine with either so I'll let you decide @pteich

Reply:
Okay. On this, it's whatever floats your boat. Not a big deal. :)

Reply:
@rgdev If even Matt is ok with it I would stick to my suggestions and use the pointer version.